include/libgit4cpp/Error.h

```diff
@@ -25,36 +25,114 @@
 #ifndef LIBGIT4CPP_ERROR_H_
 #define LIBGIT4CPP_ERROR_H_
 
-#include <stdexcept>
+#include <system_error>
 
 namespace git {
 
+class git_category_impl : public std::error_category {
```

Comment:
`<nitpick>`Should be GitCategoryImpl according to the style guide.`</nitpick>`

Reply:
But then again "our" style guide clashes with standard library customary style.
I can change that, as it is just internal.

We also had this with `SomeThing::value_type` which must be `SomeThing::ValueType` according to the style guide which is against any expectations C++ programmers will have. 

Same here, people would expect the repository type to be `git_category` because that is the form it has everywhere else, and its implementation , well, is named not `git_category_impl`, but ... `git_categoryImpl` maybe?

Luckily the guide allows to deviate when there are reasons.

Reply:
(Just rediscovered the comment above, pending, in an old browser tab. Wanted to reboot and checked all 500 open tabs :grimacing:)

Reply:
I have said my piece. But as this is internal... :shrug:

But now that I look at this: Maybe the class should not be at `namespace git` scope, but hidden even better. It could probably be an implementation detail of the `git_category()` function, couldn't it?

Reply:
I put `git_category_impl` into namespace `git::detail`.